src/main/java/com/google/devtools/build/lib/actions/ExecutionRequirements.java

```diff
@@ -119,27 +209,21 @@ public String parseIfMatches(String tag) throws ValidationException {
       ParseableRequirement.create(
           "cpu:<int>",
           Pattern.compile("cpu:(.+)"),
-          s -> {
-            Preconditions.checkNotNull(s);
-
-            int value;
-            try {
-              value = Integer.parseInt(s);
-            } catch (NumberFormatException e) {
-              return "can't be parsed as an integer";
-            }
-
-            // De-and-reserialize & compare to only allow canonical integer formats.
-            if (!Integer.toString(value).equals(s)) {
-              return "must be in canonical format (e.g. '4' instead of '+04')";
-            }
-
-            if (value < 1) {
-              return "can't be zero or negative";
-            }
-
-            return null;
-          });
+          ExecutionRequirements::IntegerValidator);
+
+  /** How many hardware threads an action requires for execution. */
+  public static final TaggedRequirement<Integer> TAGGED_CPU =
+      TaggedRequirement.create(
+          "cpu",
+          ExecutionRequirements::IntegerValidator,
+          Integer::parseInt);
+
+  /** How much real memory (resident set size) this action requires for execution. */
+  public static final TaggedRequirement<Integer> TAGGED_MEM_MB =
+      TaggedRequirement.create(
+          "memoryMb",
```

Comment:
Would it make sense to expose this as just `memory` and allow the value to take an `M` suffix or something? I realize it's more work, but would allow adding support for bigger units, which would in turn make larger numbers more readable.

Reply:
Sounds good.

Reply:
@jmmv I updated my change to require either a M or G suffix and added (hopefully) appropriate test cases.